collab-service/src/tunneller/collab_tunnel_controller.ts

```diff
@@ -104,25 +92,38 @@ class CollabTunnelController {
       new CollabTunnelSerializer(),
     );
 
+    const attemptCache = createAttemptCache();
```

Comment:
You can create the cache inside the bridge, which will reduce the amount of arguments.